Replace documentation mentions of IRC with Discord

AaronBallman opened the pull request:
This does not touch code owners or credits files that list IRC handles, that can be done separately if we want to make that change.

See https://discourse.llvm.org/t/rfc-remove-irc-as-a-recommended-communication-channel/82808/3 for the RFC.

AaronBallman committed:
Replace documentation mentions of IRC with Discord

This does not touch code owners or credits files that list IRC handles,
that can be done separately if we want to make that change.

See https://discourse.llvm.org/t/rfc-remove-irc-as-a-recommended-communication-channel/82808/3
for the RFC.

AaronBallman commented:
The developer policy got nailed by line ending changes; I'll try to redo that so it's more obvious what's changed.

erichkeane commented on llvm/docs/GettingInvolved.rst:
```diff
@@ -377,7 +377,6 @@ Guidance for office hours hosts
 
     * the `#office-hours Discord channel
```
Weird to have a list of 1.

AaronBallman replied:
fair point, I'll fix that

erichkeane commented on llvm/docs/GettingInvolved.rst:
```diff
@@ -388,27 +387,18 @@ Guidance for office hours hosts
-`invitation link <https://discord.com/invite/xS7Z362>`_.
+bot will update the channel with a link to a build bot when a build goes from
+passing to failing and again when the build goes from failing back to passing.
+It is a good channel for actively monitoring build statuses.
```
Perhaps something like, "It is a great way to actively monitor the status of the build"? 

Else this sentence reads awkward.

AaronBallman committed:
Restore the developer policy so its line endings are correct again

erichkeane reviewed (commented):
Yikes on the Dev policy, but reviewed the rest.

AaronBallman committed:
Update the developer policy

erichkeane commented on llvm/docs/DeveloperPolicy.rst:
```diff
@@ -201,7 +201,7 @@ within an area of a project:
       activities,
     * be a point of contact for contributors who need help (answering questions
-      on Discord/Discourse/IRC or holding office hours).
+      on Discord/Discourse/Discord or holding office hours).
```
Discord or more discord?

AaronBallman committed:
Remove a list of one element, move the element inline

jrtc27 reviewed (approved):
LGTM subject to the existing review comments

AaronBallman committed:
Update based on review feedback

AaronBallman committed:
Remove duplicate Discord

Remove duplicate Discord

jcranmer-intel commented on llvm/docs/GettingInvolved.rst:
```diff
@@ -388,27 +384,18 @@ Guidance for office hours hosts
-
-IRC
----
+Discord
```
You've removed the thing that lets you link to this section, if I'm recalling my RST correctly.

jcranmer-intel reviewed (commented):
One thing I have noticed is that this does seem to more or less by accident remove the IRC channel from the list of communication channels to which the code of conduct and other policies applied.

I am on board with generally replacing use of IRC with Discord, and preferring discord especially in our documentation, but I wonder if it might make sense to have some residual mention somewhere that #llvm on the OFTC network as being project-affiliated that could cover that case (even if we emphasize that it's not particularly active). I don't have any good wording suggestions, I'm afraid.

kbeyls commented on llvm/docs/CodeOfConduct.rst:
```diff
@@ -24,12 +24,12 @@ the spirit in which it's intended - a guide to make it easier to communicate
-your communication and conduct in these spaces, including emails, chats, things
-you say, slides, videos, posters, signs, or even t-shirts you display in these
-spaces. 
+LLVM Foundation. This includes Discord channels, mailing lists, bug trackers,
```
Regarding the comment on not explicitly mentioning anymore that the code of conduct continues to apply to the IRC channel too: Probably the easiest way to address that is to keep IRC in the enumerated list here.
IRC is a "legacy" channel, but it still exists, so the CoC continues to apply there.

kbeyls reviewed (commented):
Overall looks great, thank you :)
I've left one nit-picky comment on keeping IRC mentioned in the code of conduct page, as even though it's a "legacy" communication channel, it continues to exist, so the code-of-conduct continues to apply there.

AaronBallman committed:
Add IRC back to the Code of Conduct page

IRC is still a space covered by the code of conduct because it was a
legacy platform for community contributions and still has active
community member participation.

nickdesaulniers commented:
LGTM
